python/cuspatial/cuspatial/geometry/geoarrowbuffers.py

```diff
@@ -553,10 +554,17 @@ def copy(self, deep=True):
         return result
 
     def __len__(self):
-        mlength = sum(
-            self._mlines[2 * i - 1] - self._mlines[2 * i - 2] - 1
-            for i in range(1, len(self._mlines) // 2 + 1)
-        )
+        if len(self._mlines) > 0:
+            mlength = (
+                self._mlines[np.arange(1, len(self._mlines), 2)].reset_index(
+                    drop=True
+                )
+                - self._mlines[np.arange(0, len(self._mlines), 2)].reset_index(
+                    drop=True
+                )
+            ).sum() - (len(self._mlines) // 2)
```

Comment:
Is `self._mlines` a cuDF backed object?

Reply:
Ah, so, I was mistaken about them being `cudf` backed. They are usually `cudf` backed, but if we're preparing to serialize them back to `Shapely`  they'll be pandas `Series` instead of `cudf`. Do you mind if we leave this optimization out so the same code path can be used for `cudf` or `pandas`?

Reply:
Using `np.arange` is compatible with `pandas` and `cudf` `Series`, but any other method will require another code path. Do you think that's necessary? The time difference seems small...

Reply:
I think with newer versions of numpy we can use the `like` kwarg of `np.arange` to have it automatically dispatch to cupy if the object is device backed. Something like:
```
np.arange(0, len(self._mlines), 2, like=self._mlines.values)
```

Reply:
not using `.take` will force materializing a `RangeIndex` and then gathering it as well. If the index is meaningless here maybe we could just operate on `.values` instead to work with arrays?

Reply:
.values works with either pandas or cudf, I've done that.

Reply:
Anything else? Let me know.